File: OC.UI/src/app/blci/api/locations/region-leadership.ts
import { LeadershipRole } from "../members/leadership-role";
import { Member } from "../members/member";
import { Region } from "./region";

export interface RegionLeadership {
    id?: number;
    regionId?: number;
    regionName?: string;
    leadershipRoleId?: number;
    leadershipRoleName?: string;
    memberId?: number;
    memberName?: string;
    memberPictureUrl?: string;
    isActive?: boolean;
    region?: Region;
    leadershipRole?: LeadershipRole;
    member?: Member;
}

File: OC.Domain/Models/Locations/RegionLeadership.cs
using OC.Domain.Interfaces;
using OC.Domain.Models.Members;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OC.Domain.Models.Locations
{
    public class RegionLeadership : AuditableEntity, IEntityBase
    {
        public long Id { get; set; }
        public long RegionId { get; set; }
        public long LeadershipRoleId { get; set; }
        public long MemberId { get; set; }

        // Relationships
        public virtual Region? Region { get; set; }
        public virtual LeadershipRole? LeadershipRole { get; set; }
        public virtual Member? Member { get; set; }
    }
}
